internal/configs/virtualserver.go

```diff
@@ -1474,18 +1474,26 @@ func generateLocation(path string, upstreamName string, upstream conf_v1.Upstrea
 }
 
 func generateProxySetHeaders(proxy *conf_v1.ActionProxy) []version2.Header {
-	if proxy == nil || proxy.RequestHeaders == nil {
-		return nil
-	}
-
 	var headers []version2.Header
-	for _, h := range proxy.RequestHeaders.Set {
-		headers = append(headers, version2.Header{
-			Name:  h.Name,
-			Value: h.Value,
-		})
+
+	hostHeaderValue := "$host"
```

Comment:
This code is unclear. I think you're trying to add a default host header with a value of '$host', if no such header exists. 

```
hasHostHeader := false

if proxy != nil && proxy.RequestHeaders != nil {
     for _, h := range proxy.RequestHeaders.Set {
 	    headers = append(headers, version2.Header{
 		    Name:  h.Name,
 		    Value: h.Value,
 	    })
 	    if strings.ToLower(h.Name) == "host" {
 		    hasHostHeader = true
 	    }
     }
}


if !hasHostHeader {
     headers = append(headers, version2.Header{Name: "Host", Value: "$host"})
}
 	
```

Reply:
@soneillf5 
the intention was also to prevent setting duplicate host headers, if a user configures that. In that case, the last one in the list would win.

however, since we didn't report any warning/error in that case and ignored the duplicate(s) silently, I think it's better to allow it(them) and make it a user error. Such configuration is valid from NGINX perspective. 

I used your suggestion.

Reply:
Brilliant, thanks.